backend/src/dao/CoffeeChatDao.ts

```diff
@@ -0,0 +1,115 @@
+import { v4 as uuidv4 } from 'uuid';
+import { db, memberCollection, coffeeChatsCollection } from '../firebase';
+import { DBCoffeeChat } from '../types/DataTypes';
+import { getMemberFromDocumentReference } from '../utils/memberUtil';
+import BaseDao from './BaseDao';
+import { deleteCollection } from '../utils/firebase-utils';
+
+async function materializeCoffeeChat(dbCoffeeChat: DBCoffeeChat): Promise<CoffeeChat> {
+  const member1 = await getMemberFromDocumentReference(dbCoffeeChat.members[0]);
+  const member2 = await getMemberFromDocumentReference(dbCoffeeChat.members[1]);
+
+  return {
+    ...dbCoffeeChat,
+    members: [member1, member2]
+  };
+}
+
+async function serializeCoffeeChat(coffeeChat: CoffeeChat): Promise<DBCoffeeChat> {
+  const member1Data = await memberCollection.doc(coffeeChat.members[0].email);
+  const member2Data = await memberCollection.doc(coffeeChat.members[1].email);
+
+  return {
+    ...coffeeChat,
+    members: [member1Data, member2Data]
+  };
+}
+
+export default class CoffeeChatDao extends BaseDao<CoffeeChat, DBCoffeeChat> {
+  constructor() {
+    super(coffeeChatsCollection, materializeCoffeeChat, serializeCoffeeChat);
+  }
+
+  /**
+   * Creates a new coffee chat for member
+   * @param coffeeChat - Newly created CoffeeChat object.
+   * If provided, the object uuid will be used. If not, a new one will be generated.
+   * The pending field will be set to true by default.
+   * A member can not create a coffee chat the same person from previous semesters
+   */
+
+  async createCoffeeChat(coffeeChat: CoffeeChat): Promise<CoffeeChat> {
+    const [member1, member2] = coffeeChat.members;
+
+    if (member1.netid === member2.netid) {
+      throw new Error(`Cannot create coffee chat with yourself.`);
+    }
+    const prevChats1 = await this.getCoffeeChatsByUser(member1);
+    const prevChats2 = await this.getCoffeeChatsByUser(member2);
+    const prevChats = [...prevChats1, ...prevChats2];
+
+    if (prevChats.some((c) => c.members.includes(member1) && c.members.includes(member2))) {
```

Comment:
I would be very careful about this check. `includes(...)` uses object equality and we've had a bug introduced in candidate decider in the past from this. Consider instead checking against netIDs or emails.

Reply:
Alternatively, we use lodash's `isequal` which checks for deep equality for the profile update notification job. That can be used here.